sql-plugin/src/main/scala/com/nvidia/spark/rapids/TypeChecks.scala

```diff
@@ -2193,15 +2211,119 @@ object SupportedOpsForTools {
+        }.toMap
+
+        val notes = execChecks.supportNotes
+        val inputs = allData.values.head.keys
+
+        val firstCol = Seq(rule.tag.runtimeClass.getSimpleName)
+        val thirdCol = Seq(rule.notes().getOrElse("None"))
+        inputs.foreach { input =>
+          val named = notes.get(input)
+              .map(l => input + "(" + l.mkString(";") + ")")
+              .getOrElse(input)
+          val supportLevelOps = allSupportedTypes.toSeq.map { t =>
+            allData(t)(input).text
+          }
+          val isSupportedExec = Seq(if (supportLevelOps.forall(_.equals("NS"))) "NS" else "S")
+          val allCols = (firstCol ++ isSupportedExec ++ thirdCol ++ Seq(named) ++ supportLevelOps)
+          println(s"${allCols.map(replaceDelimiter(_, ",")).mkString(",")}")
+        }
+      }
+    }
+  }
+
+  private def outputSupportedExpressions(): Unit = {
+    // TODO Look at what we have for defaults for some configs because if the configs are off
+    // it likely means something isn't completely compatible.
+    val conf = new RapidsConf(Map.empty[String, String])
+    val types = allSupportedTypes.toSeq
+    val header = Seq("Expression", "Supported", "SQL Func", "Notes", "Context", "Params") ++ types
+    println(header.mkString(","))
+    GpuOverrides.expressions.values.toSeq.sortBy(_.tag.toString).foreach { rule =>
+      val checks = rule.getChecks
+      if (rule.isVisible && checks.isDefined && checks.forall(_.shown)) {
+        val sqlFunctions =
+          ConfHelper.getSqlFunctionsForClass(rule.tag.runtimeClass).map(_.mkString(", "))
+        val exprChecks = checks.get.asInstanceOf[ExprChecks]
+        // Params can change between contexts, but should not
+        val allData = allSupportedTypes.map { t =>
+          (t, exprChecks.support(t))
+        }.toMap
+        val representative = allData.values.head
+        val firstCol = Seq(rule.tag.runtimeClass.getSimpleName)
+        val staticCols = Seq(sqlFunctions.getOrElse(" "), rule.notes().getOrElse("None"))
+
+        representative.foreach {
+          case (context, data) =>
+            data.keys.foreach { param =>
+              val supportLevelOps = allSupportedTypes.toSeq.map { t =>
+                allData(t)(context)(param).text
+              }
+              val isSupportedExec = Seq(if (supportLevelOps.forall(_.equals("NS"))) "NS" else "S")
+              val allCols = (firstCol ++ isSupportedExec ++ staticCols ++ Seq(context.toString)
+                  ++ Seq(param) ++ supportLevelOps)
+              println(s"${allCols.map(replaceDelimiter(_, ",")).mkString(",")}")
+            }
+        }
+      }
+    }
+  }
+
+  def help(printType: String): Unit = {
+    printType match {
+      case a if a.equals("execs") => outputSupportedExecs()
+      case expr if (expr.equals("exprs")) => outputSupportedExpressions()
+      case score if (score.equals("operatorScore")) => operatorMappingWithScore()
+      case _ => outputSupportIO()
```

Comment:
change to match on "ioOnly" and then add a default case which throws an error and prints out valid options.